File: pm_core/src/infrastructure/store/model/priority_scheme_model.go
/*
Author: QuanTuanHuy
Description: Part of Serp Project
*/

package model

type PrioritySchemeModel struct {
	BaseModel
	TenantID          int64   `gorm:"column:tenant_id;not null" json:"tenantId"`
	Name              string  `gorm:"column:name;type:varchar(255);not null" json:"name"`
	Description       *string `gorm:"column:description" json:"description,omitempty"`
	DefaultPriorityID int64   `gorm:"column:default_priority_id;not null" json:"defaultPriorityId"`
}

func (PrioritySchemeModel) TableName() string {
	return "priority_schemes"
}

File: pm_core/src/main/java/serp/project/pmcore/infrastructure/store/model/PrioritySchemeModel.java
/**
 * Author: QuanTuanHuy
 * Description: Part of Serp Project
 */

package serp.project.pmcore.infrastructure.store.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.SuperBuilder;
import org.hibernate.annotations.SQLRestriction;

@Entity
@Table(name = "priority_schemes")
@SQLRestriction("deleted_at IS NULL")
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@SuperBuilder
public class PrioritySchemeModel extends BaseModel {

    @Column(name = "tenant_id", nullable = false)
    private Long tenantId;

    @Column(name = "name", nullable = false)
    private String name;

    @Column(name = "description")
    private String description;

    @Column(name = "default_priority_id")
    private Long defaultPriorityId;
}
